composer/profiler/profiler.py

```diff
@@ -167,20 +152,45 @@ class Profiler:
         repeat (int, optional): Number of profiling cycles to perform per epoch. Set to ``0`` to record the entire epoch. (Default: ``1``)
     """
 
-    def __init__(
-        self,
-        state: State,
-        event_handlers: Sequence[ProfilerEventHandler],
-        skip_first: int = 0,
-        wait: int = 0,
-        warmup: int = 1,
-        active: int = 4,
-        repeat: int = 1,
-        merged_trace_file: str = "merged_profiler_trace.json",
-    ) -> None:
+    def __init__(self,
+                 event_handlers: Sequence[ProfilerEventHandler] = None,
+                 skip_first: int = 0,
+                 wait: int = 0,
+                 warmup: int = 1,
+                 active: int = 4,
+                 repeat: int = 1,
+                 merged_trace_file: str = "merged_profiler_trace.json",
+                 profile_cpu: bool = True,
+                 profile_sys_memory: bool = False,
+                 profile_disk: bool = False,
+                 profile_net: bool = False,
+                 stats_thread_interval_seconds: float = 0.5,
+                 torch_profiling: bool = True,
```

Comment:
Can you remove this parameter, and instead make `tensorboard_trace_handler_dir` optional and default to None? If the `tensorboard_trace_handler_dir` is None, then no torch profiling -- otherwise, if it is set, then traces will be stored there.